R/gh.R

```diff
@@ -234,21 +233,84 @@ gh_response_length <- function(res) {
   }
 }
 
-gh_make_request <- function(x) {
-  method_fun <- list(
-    "GET" = GET, "POST" = POST, "PATCH" = PATCH,
-    "PUT" = PUT, "DELETE" = DELETE
-  )[[x$method]]
-  if (is.null(method_fun)) {
+gh_make_request <- function(x, error_call = caller_env()) {
+  if (!x$method %in% c("GET", "POST", "PATCH", "PUT", "DELETE")) {
     cli::cli_abort("Unknown HTTP verb: {.val {x$method}}")
   }
 
-  raw <- do.call(
-    method_fun,
-    compact(list(
-      url = x$url, query = x$query, body = x$body,
-      add_headers(x$headers), x$dest
-    ))
+  req <- httr2::request(x$url)
+  req <- httr2::req_method(req, x$method)
+  req <- httr2::req_url_query(req, !!!x$query)
+  if (is.raw(x$body)) {
+    req <- httr2::req_body_raw(req, x$body)
+  } else {
+    req <- httr2::req_body_json(req, x$body, null = "list", digits = 4)
+  }
+  req <- httr2::req_headers(req, !!!x$headers)
+
+  # use retry-after info when possible
+  # https://docs.github.com/en/rest/overview/resources-in-the-rest-api#exceeding-the-rate-limit
+  github_is_transient <- function(resp) {
```

Comment:
Should we make this behavior configurable? In some situations, I don't want to wait an hour.

Reply:
Yeah, I wondered about that. And maybe we need some support for https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#secondary-rate-limits too?